site/src/sphinx/client-retry.rst

```diff
@@ -152,32 +158,63 @@ using :api:`RetryingHttpClientBuilder`:
-                if (thrown != null) {
-                    if (thrown instanceof ResponseTimeoutException ||
-                        thrown instanceof UnprocessedRequestException) {
-                        // The response timed out or the request has not been handled by the server.
+    final RetryStrategyWithContent<HttpResponse> strategy =
+        new RetryStrategyWithContent<HttpResponse>() {
+
+            final Backoff backoff = Backoff.ofDefault();
+
+            @Override
+            public CompletionStage<Backoff> shouldRetry(ClientRequestContext ctx,
+                                                        HttpResponse response) {
+                return response.aggregate().handle((result, thrown) -> {
+                    if (thrown != null) {
+                        if (thrown instanceof ResponseTimeoutException ||
+                            thrown instanceof UnprocessedRequestException) {
+                            // The response timed out or the request has not been handled
+                            // by the server.
+                            return backoff;
+                        }
+                    } else if ("Should I retry?".equals(result.contentUtf8())) {
                         return backoff;
                     }
-                } else if ("Should I retry?".equals(result.contentUtf8())) {
-                    return backoff;
-                }
-                return null; // Return null to stop retrying.
-            });
-        }
-    };
+                    return null; // Return null to stop retrying.
+                });
+            }
+        };
 
+    // Create an HttpClient with a custom strategy.
     final HttpClient client = new HttpClientBuilder(...)
-            .decorator(new RetryingHttpClientBuilder(strategy).newDecorator()) // Specify the strategy.
+            .decorator(new RetryingHttpClientBuilder(strategy).newDecorator())
             .build();
 
     final AggregatedHttpMessage res = client.execute(...).aggregate().join();
 
+.. tip::
+
+    You might find the ``peel()`` method in :api:`Exceptions` useful when the exception you are trying to
+    handle is wrapped by exceptions like ``CompletionException`` and ``ExecutionException``:
+
+    .. code-block:: java
+
+        import com.linecorp.armeria.common.Exceptions;
+
+        @Override
+        public CompletionStage<Backoff> shouldRetry(ClientRequestContext ctx,
+                                                    @Nullable Throwable cause) {
+            if (cause != null) {
+                if (cause instanceof ResponseTimeoutException ||
+                    cause instanceof UnprocessedRequestException) {
+                    // The response timed out or the request has not been handled
+                    // by the server.
+                    return CompletableFuture.completedFuture(backoff);
+                }
+
+                Throwable peeled = Exceptions.peel(cause);
```

Comment:
How about moving this right under the `if (cause != null) {`? or Might we remove the `if` condition which checks the exception instance because it's already described in the above example?